src/content/docs/en/recipes/modified-time.mdx

```diff
@@ -0,0 +1,149 @@
+---
+title: Add Last Modified Time
+description: Build a remark plugin to add the last modified time to your Markdown and MDX.
+i18nReady: false
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+Learn how to build a [remark plugin](https://github.com/remarkjs/remark) that adds the last modified time to the frontmatter of your Markdown and MDX files. Use this property to  display the modified time in your pages.
+
+## Recipe
+
+1. Install Helper Packages
+
+   Install [`Day.js`](https://www.npmjs.com/package/dayjs) to modify and format times:
+
+   <PackageManagerTabs>
+     <Fragment slot="npm">
+       ```shell
+       npm install dayjs
+       ```
+     </Fragment>
+     <Fragment slot="pnpm">
+       ```shell
+       pnpm install dayjs
+       ```
+     </Fragment>
+     <Fragment slot="yarn">
+       ```shell
+       yarn add dayjs
+       ```
+     </Fragment>
+   </PackageManagerTabs>
+
+2. Create a Remark Plugin
+
+   There are two ways to get the modified time of a file; using Git, or using the filesystem.
+
+   :::note
+   It's highly recommended to use Git to get the modified time of the files. Git operations (like cloning) will result in the file system modified time being updated.
+   :::
+
+   - Using Git
+
+     This plugin uses `execSync` to run a git command that returns the commit timestamp in ISO 8601 format. The timestamp is then added to the frontmatter of the file.
+
+     ```js
+     import { execSync } from "child_process";
+
+     export function modifiedTime() {
+       return function (tree, file) {
+         const filepath = file.history[0];
+         const result = execSync(`git log -1 --pretty="format:%cI" ${filepath}`);
+         file.data.astro.frontmatter.lastModified = result.toString();
+       };
+     }
+     ```
+
+   - Using File System
+
+     This plugin uses `statSync` to get the `mtime` (modified time) of the file in ISO 8601 format. The timestamp is then added to the frontmatter of the file.
+     ```js
+     import { statSync } from "fs";
+
+     export function modifiedTime() {
+       return function (tree, file) {
+         const filepath = file.history[0];
+         const result = statSync(filepath);
+         file.data.astro.frontmatter.lastModified = result.mtime.toISOString();
+       };
+     }
+     ```
```

Comment:
``````suggestion
2. Create a Remark Plugin

   This plugin uses `execSync` to run a git command that returns the timestamp of the latest commit in ISO 8601 format. The timestamp is then added to the frontmatter of the file.

     ```js
     import { execSync } from "child_process";

     export function modifiedTime() {
       return function (tree, file) {
         const filepath = file.history[0];
         const result = execSync(`git log -1 --pretty="format:%cI" ${filepath}`);
         file.data.astro.frontmatter.lastModified = result.toString();
       };
     }
     ```
		<details>
		<summary>Using the file system instead of git</summary>
			Although using git is the recommended way to get the last modified timestamp from a file, it is possible to use the file system directly. 
     This plugin uses `statSync` to get the `mtime` (modified time) of the file in ISO 8601 format. The timestamp is then added to the frontmatter of the file.
     ```js
     import { statSync } from "fs";

     export function modifiedTime() {
       return function (tree, file) {
         const filepath = file.history[0];
         const result = statSync(filepath);
         file.data.astro.frontmatter.lastModified = result.mtime.toISOString();
       };
     }
     ```
			</details>
``````

So, this formatting may end up being off, but when one version is *strongly recommended*, it's often better to just show that one. (And, we don't like Notes when we don't really need them.)

So, what about something like *this*? (I'll want to see it in the preview to be sure, but I think this might be a good way of making it clear which one is recommended, and not putting extra info for the people who aren't going to follow your recommendation.)

Reply:
Looks good! I'll take a look at some formatting tweaks.